bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/TestBigtableDataGrpcClient.java

```diff
@@ -265,6 +276,105 @@ public void testScannerIdle() throws IOException {
+
+    // Start the call
+    ListenableFuture<List<FlatRow>> resultFuture =
+        defaultClient.readFlatRowsAsync(requestBuilder.build());
+
+    // Capture the caller's listener
+    ArgumentCaptor<ClientCall.Listener> listenerCaptor =
+        ArgumentCaptor.forClass(ClientCall.Listener.class);
+    verify(mockClientCall, times(1)).start(listenerCaptor.capture(), any(Metadata.class));
+
+    // Get the listener for the first attempt
+    Listener listener = listenerCaptor.getValue();
+    listener.onMessage(
+        RetryingReadRowsOperationTest.buildResponse(ByteString.copyFromUtf8("Key1")));
+    listener.onClose(
+        Status.CANCELLED.withCause(
+            new Watchdog.StreamWaitTimeoutException(State.WAITING, TimeUnit.MINUTES.toMillis(10))),
+        new Metadata());
+
+    // Verify that the retry was scheduled
+    ArgumentCaptor<Runnable> scheduledRetryCaptor = ArgumentCaptor.forClass(Runnable.class);
+    verify(mochScheduler, times(1))
+        .schedule(scheduledRetryCaptor.capture(), anyLong(), any(TimeUnit.class));
+    Runnable scheduledRetry = scheduledRetryCaptor.getValue();
+    scheduledRetry.run();
+
+    // Get the listener for the retry attempyt
+    listener = listenerCaptor.getValue();
+    listener.onMessage(
+        RetryingReadRowsOperationTest.buildResponse(ByteString.copyFromUtf8("Key2")));
+    listener.onClose(Status.OK, new Metadata());
+
+    Assert.assertEquals(
+        resultFuture.get(),
+        RowMerger.toRows(
+            Lists.newArrayList(
+                RetryingReadRowsOperationTest.buildResponse(
+                    ByteString.copyFromUtf8("Key1"), ByteString.copyFromUtf8("Key2")))));
+  }
+
+  @Test
+  public void testReadFlatRowsAsyncWaitTimeoutRetryFailEventually() throws Exception {
+    // Run retries immediately
+    Mockito.when(mochScheduler.schedule(any(Runnable.class), anyLong(), any(TimeUnit.class)))
+        .thenAnswer(
+            new Answer<ScheduledFuture>() {
+              @Override
+              public ScheduledFuture answer(InvocationOnMock invocation) {
+                Runnable runnable = invocation.getArgument(0);
+                runnable.run();
+                return null;
+              }
+            });
+
+    ReadRowsRequest.Builder requestBuilder = ReadRowsRequest.newBuilder().setTableName(TABLE_NAME);
+
+    // Start the call
+    ListenableFuture<List<FlatRow>> resultFuture =
+        defaultClient.readFlatRowsAsync(requestBuilder.build());
+
+    for (int i = 0; i < 10; i++) {
+      // Get the caller's listener for the current attempt.
+      ArgumentCaptor<ClientCall.Listener> listenerCaptor =
+          ArgumentCaptor.forClass(ClientCall.Listener.class);
+      verify(mockClientCall, times(1)).start(listenerCaptor.capture(), any(Metadata.class));
+      Listener listener = listenerCaptor.getValue();
+
+      // Reset mock before starting the next attempt
+      Mockito.reset(mockClientCall);
```

Comment:
Why are we resetting the mock? The answer might be a better comment here.

Reply:
I added some comments to explain